### ISSUE TITLE
How to gracefully handle the case where the guide is in a finished state

### ISSUE THREAD

**Issue by rlouf** [rlouf/dotllm#2]
We currently return the logits unmodified when the guide is finished. It leads to the right completion nevertheless, so I need to figure out how the sampling process works internally with vLLM.
